### ISSUE TITLE
gpu-context should auto switch when switch hwdec

### ISSUE THREAD

**Issue by Andarwinux** [mpv-player/mpv#13926]
Before requesting a new feature make sure it hasn't been requested yet.
https://github.com/mpv-player/mpv/labels/meta%3Afeature-request

### Expected behavior of the wanted feature
Example: 
```
gpu-context=d3d11,winvk
hwdec=d3d11va
```
Then switch hwdec to nvdec at runtime (`set hwdec nvdec`) , then `gpu-context` should auto switch to winvk to allow `hwdec=nvdec` to work without error.

**Comment by kasper93** [comment]
@na-na-hi What do you think?

**Comment by Hrxn** [comment]
`hwdec=nvdec` requires `gpu-context=winvk` now? At least on Windows. Is this by design?

**Comment by philipl** [comment]
There are interops for egl and vulkan, so it doesn't have to be winvk but that is the better choice. It is possible to write a d3d11 interop but someone has to do the work

**Comment by Hrxn** [comment]
I see.. So, not some recent change mandated by newer versions of NVIDIA's Windows drivers?

**Comment by philipl** [comment]
No. Egl should still work fine but I haven't touched windows for ages.
